symcrypt-sys/build/static_link.rs

```diff
@@ -0,0 +1,368 @@
+ghash.c
+hash.c
+hkdf_selftest.c
+hkdf.c
+hmac.c
+hmacmd5.c
+hmacsha1.c
+hmacsha224.c
+hmacsha256.c
+hmacsha384.c
+hmacsha512.c
+hmacsha512_224.c
+hmacsha512_256.c
+hmacsha3_224.c
+hmacsha3_256.c
+hmacsha3_384.c
+hmacsha3_512.c
+kmac.c
+libmain.c
+lms.c
+marvin32.c
+md2.c
+md4.c
+md5.c
+mldsa.c
+mldsa_primitives.c
+mlkem.c
+mlkem_primitives.c
+modexp.c
+paddingPkcs7.c
+parhash.c
+pbkdf2_hmacsha1.c
+pbkdf2_hmacsha256.c
+pbkdf2.c
+poly1305.c
+primes.c
+rc2.c
+rc4.c
+rdrand.c
+rdseed.c
+recoding.c
+rsa_enc.c
+rsa_padding.c
+rsakey.c
+ScsTable.c
+scsTools.c
+selftest.c
+service_indicator.c
+session.c
+sha1.c
+sha256.c
+sha256Par.c
+sha256Par-ymm.c
+sha256-xmm.c
+sha256-ymm.c
+sha512.c
+sha512Par.c
+sha512Par-ymm.c
+sha512-ymm.c
+sha3.c
+sha3_224.c
+sha3_256.c
+sha3_384.c
+sha3_512.c
+shake.c
+sp800_108_hmacsha1.c
+sp800_108_hmacsha256.c
+sp800_108_hmacsha512.c
+sp800_108.c
+srtp_kdf.c
+srtp_kdf_selftest.c
+ssh_kdf.c
+ssh_kdf_sha256.c
+ssh_kdf_sha512.c
+sskdf.c
+sskdf_selftest.c
+tlsCbcVerify.c
+tlsprf_selftest.c
+tlsprf.c
+xmss.c
+xtsaes.c
+";
+
+// only for x86_64_unknown_linux_gnu
+const SPECIAL_FLAGS: &str = r#"
+set_source_files_properties(aes-ymm.c PROPERTIES COMPILE_OPTIONS "-mavx;-mavx2;-mvaes;-mvpclmulqdq")
+set_source_files_properties(sha256Par-ymm.c PROPERTIES COMPILE_OPTIONS "-mavx;-mavx2")
+set_source_files_properties(sha512Par-ymm.c PROPERTIES COMPILE_OPTIONS "-mavx;-mavx2")
+set_source_files_properties(sha256-xmm.c PROPERTIES COMPILE_OPTIONS "-mssse3")
+set_source_files_properties(sha256-ymm.c PROPERTIES COMPILE_OPTIONS "-mavx;-mavx2;-mbmi2")
+set_source_files_properties(sha512-ymm.c PROPERTIES COMPILE_OPTIONS "-mavx;-mavx2;-mbmi2")
+"#;
+
+fn compile_symcrypt_static(lib_name: &str, options: &SymCryptOptions) -> std::io::Result<()> {
+    let (already_compiled_files, intermediates) = compile_intermediates(&options);
+
+    let mut base_files: Vec<&'static str> = CMAKE_SOURCES_COMMON
+        .lines()
+        .filter(|line| {
+            let line = line.trim();
+            !(line.is_empty() || line.starts_with("#") || already_compiled_files.contains(&line))
+        })
+        .collect();
+
+    base_files.push("env_generic.c"); // symcrypt_generic
+
+    let mut module_files = vec![];
+
+    match options.triple() {
+        Triple::x86_64_pc_windows_msvc | Triple::aarch64_pc_windows_msvc => {
+            base_files.push("env_windowsUserModeWin8_1.c");
+            base_files.push("IEEE802_11SaeCustom.c");
+            module_files.push("inc/static_WindowsDefault.c");
+        }
+        Triple::x86_64_unknown_linux_gnu => {
+            base_files.push("linux/intrinsics.c");
+            base_files.push("env_posixUserMode.c");
+            module_files.push("inc/static_LinuxDefault.c");
+        }
+        Triple::aarch64_unknown_linux_gnu => {
+            base_files.push("env_posixUserMode.c");
+            module_files.push("inc/static_LinuxDefault.c");
+        }
+    }
+
+    let asm_files = match options.triple() {
```

Comment:
The required `.asm` files would ideally be checked in and maintained by SymCrypt